server/repositories/cmsApi.js

```diff
@@ -71,6 +76,18 @@ class CmsApi {
     });
   }
 
+  async getCache(query) {
+    const key = query.getKey?.() || null;
+    const expiry = query.getExpiry?.() || 8640;
+    let data = key ? await this.#cache.get(key) : null;
+    if (!data) {
+      data = (await this.get(query)) || [];
+      if (key && expiry && data) await this.#cache.set(key, data, expiry);
```

Comment:
key and expiry won't ever be null will they?

Reply:
Key theoretically might be - but thats an error, so I probably could/should throw an error in that scenario instead 🤔

Reply:
but data would then be null?

Reply:
I threw an error for missing keys.